PythonVisualizations/Heap.py

```diff
@@ -0,0 +1,518 @@
+        self.cleanUp(callEnviron)
+
+    # makes a new heap of size 2
+    def newArray(self):
+        # gets rid of old elements in the list
+        del self.list[:]
+        self.size = 2
+        self.display()
+        # make a new arrow pointing to the top of the Heap
+        self.indexDisplay = self.createIndex(len(self.list))
+
+    def peek(self):
+        self.startAnimations()
+        callEnviron = self.createCallEnvironment()
+
+        # draw output box
+        canvasDimensions = self.widgetDimensions(self.canvas)
+        spacing = self.CELL_WIDTH * 3 // 4
+        padding = 10
+        boxSize = 1
+        peekIndex = 0
+        # changed the values that were added to the variables to get the output box in a desirable location
+        outputBox = self.canvas.create_rectangle(
+            (self.HEAP_X0 * 2 + padding),
+            canvasDimensions[1] - self.CELL_WIDTH - padding,
+            (self.HEAP_X0 * 2 + boxSize * spacing + padding * 2),
+            canvasDimensions[1] - padding,
+            fill=self.OPERATIONS_BG)
+        callEnviron.add(outputBox)
+
+        outputBoxCoords = self.canvas.coords(outputBox)
+        midOutputBox = (outputBoxCoords[3] + outputBoxCoords[1]) // 2
+        # create the value to move to output box
+        valueOutput = self.copyCanvasItem(self.list[peekIndex].display_val)
+        valueList = (valueOutput,)
+        callEnviron.add(valueOutput)
+        # move value to output box
+        # self.highlightCodeTags('print', callEnviron)
+        toPositions = (outputBoxCoords[0] + padding / 2 + (peekIndex + 1 / 2) * spacing,
+                        midOutputBox)
+        self.moveItemsTo(valueList, (toPositions,), sleepTime=.02)
+        # make the value 25% smaller
+        newSize = (self.VALUE_FONT[0], int(self.VALUE_FONT[1] * .75))
+        self.canvas.itemconfig(valueOutput, font=newSize)
+
+        # add label to output box
+        labelX = self.HEAP_X0 * 2 + boxSize * spacing + padding * 5
+        labelY = canvasDimensions[1] - padding * 2
+
+        boxLabel = self.canvas.create_text(
+            labelX, labelY, text="root", font=self.VARIABLE_FONT)
+        callEnviron.add(boxLabel)
+        self.highlightCodeTags([], callEnviron)
+        # finish animation
+        self.cleanUp(callEnviron)
+
+
+    def removeMax(self):
+        self.startAnimations()
+        callEnviron = self.createCallEnvironment()
+
+        # remove first element from list
+        n = self.list[0]
+        self.list[0] = None
+        self.size -= 1
+        # if the array is now empty, delete the NoneType place holder from the array
+        if self.size < 1: del self.list[:]
+        # Slide value rectangle up and off screen
+        items = (n.display_shape, n.display_val)
+        toPositions = (100, -100, 100, -50)
+        callEnviron |= set(items)
+        self.moveItemsBy(items, delta=(0, -max(400, self.canvas.coords(n.display_shape)[3])), steps=self.CELL_HEIGHT,

```

Comment:
Assuming the `n.display_shape` is visible on the canvas, it's maximum coordinate value is 400 (with the default canvas size).  I don't think you need the `max` in determining the delta.  You could just use `delta=(0, -self.canvas.coords(n.display_shape)[3])`